bundle/config/bundle.go

```diff
@@ -18,6 +18,9 @@ type Bundle struct {
 	// Target is set by the mutator that selects the target.
 	Target string `json:"target,omitempty" bundle:"readonly"`
 
+	// TargetConfig stores a snapshot of the target configuration when it was selected by SelectTarget.
+	TargetConfig *Target `json:"target_config,omitempty" bundle:"internal"`
+
```

Comment:
If located here, it is a part of the configuration, therefore the schema, the output of validate, etc.

Please move this to be a property on the `bundle.Bundle` type to avoid this.

Reply:
👍 Ah, yes, you're right. I moved it.

Do you still think it's the right balance to have that extra property there. It's an extra field that copies the bundle config but it only has a value depending on which state the deployment is in. I'm okay with the proposed approach but it seems cleaner to me to have fewer properties and to do a cleanup step just before summary/validate.